### PR TITLE
Web: return user traits with getUser request

### PR DESCRIPTION
**Pull request by kimlisa**
part of https://github.com/gravitational/teleport.e/issues/1899 where we calculate eligible users from a list of users based on required roles and traits defined in an access list

### CODE REVIEW

File: lib/web/ui/user.go [hunk 1]
```diff
@@ -30,6 +30,8 @@ type UserListEntry struct {
 	// AuthType is the type of auth service
 	// that the user was authenticated through.
 	AuthType string `json:"authType"`
+	// RoleVariableTraits are traits used for role variables.
+	RoleVariableTraits map[string][]string `json:"roleVariableTraits`
```
Comment (kimlisa) [inline lib/web/ui/user.go:34]: we already have a field called `trait`, (line 37 of this file), but these traits refer to allowed resource logins/usernames/resourcenames/roles

**Review by ryanclark — APPROVED** [review]

File: lib/web/ui/user.go (hunk 1 above)
Comment (mdwn) [reply lib/web/ui/user.go]: I can't think of a better name, but I'm not sure that this name accurately describes the user's traits here. Maybe LoginTraits or something like that?

Comment (kimlisa) [reply lib/web/ui/user.go]: i just renamed to `allTraits`... is that better? 

```
        // AllTraits returns all the traits.
	// Different from "userTraits" where "userTraits"
	// "selectively" defines traits for local accounts.
	AllTraits map[string][]string `json:"allTraits"`
```

Comment (kimlisa) [reply lib/web/ui/user.go]: as discussed in slack, renamed to `externalTraits`

File: web/packages/teleport/src/services/user/makeUser.ts [hunk 2]
```diff
@@ -18,7 +18,7 @@ import { User } from './types';
 
 export default function makeUser(json: any): User {
   json = json || {};
-  const { name, roles, authType, traits = {} } = json;
+  const { name, roles, authType, traits = {}, roleVariableTraits } = json;
```
Comment (mdwn) [inline web/packages/teleport/src/services/user/makeUser.ts:21]: Update to externalTraits.

File: web/packages/teleport/src/services/user/makeUser.ts [hunk 3]
```diff
@@ -34,6 +34,7 @@ export default function makeUser(json: any): User {
       windowsLogins: traits.windowsLogins || [],
       awsRoleArns: traits.awsRoleArns || [],
     },
+    roleVariableTraits: roleVariableTraits || {},
```
Comment (mdwn) [inline web/packages/teleport/src/services/user/makeUser.ts:37]: Update to externalTraits.

File: web/packages/teleport/src/services/user/types.ts [hunk 4]
```diff
@@ -95,6 +97,7 @@ export interface User {
   // isLocal is true if json.authType was 'local'.
   isLocal?: boolean;
   traits?: UserTraits;
+  roleVariableTraits?: RoleVariableTraits;
```
Comment (mdwn) [inline web/packages/teleport/src/services/user/types.ts:100]: Update to externalTraits.

File: web/packages/teleport/src/services/user/user.test.ts [hunk 5]
```diff
@@ -270,6 +270,7 @@ test('fetch users, null response values gives empty array', async () => {
       isLocal: false,
       name: '',
       roles: [],
+      roleVariableTraits: {},
```
Comment (mdwn) [inline web/packages/teleport/src/services/user/user.test.ts:273]: Update to externalTraits.

**Review by mdwn — APPROVED** [review]
There are a few fields that need renaming, but otherwise LGTM.

**Pull request merged**